determined-ai/determined
# feat: Connect ProjectStore with streaming updates

## Description


Connects workspace details page to ProjectStore, and ProjectStore monitors projects changes through streaming updates after the initial fetch. 


## Test Plan



- Navigate to workspace details page, Inspect -> Network -> WS and there should be a webSocket for "/stream"
- Sorting should work as expected
- Change the name or archive a project, the change should be reflected very quickly and note that there is no API call
- Open 2 browsers with 2 different workspace detail page, move project from one workspace to another, the project should move as expected and there should be no API call


## Commentary (optional)


Note that if viewing as list, the changes of description is not immediately reflected due to how the Input is setup. 


## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

[MD-265](https://hpe-aiatscale.atlassian.net/browse/MD-265)




[MD-265]: https://hpe-aiatscale.atlassian.net/browse/MD-265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
